rust/src/delta.rs

```diff
@@ -1277,12 +1277,7 @@ impl DeltaTable {
         if dry_run {
             return Ok(files_to_delete);
         }
-
-        let paths = &files_to_delete
-            .iter()
-            .map(|rel_path| self.storage.join_path(&self.table_uri, rel_path))
-            .collect::<Vec<_>>();
-        match self.storage.delete_objs(paths).await {
+        match self.storage.delete_objs(&files_to_delete).await {
```

Comment:
I'm not sure how this was working in object stores, but in local fs `files_to_delete` was already absolute. Local file system used `Path::join()`, which had the fortunate behavior of just using the second path if it was already absolute.

Reply:
well... maybe its not :) - will have a closer look once i get to #647.